test/UnichainProtocolFees.fork.t.sol

```diff
@@ -0,0 +1,217 @@
+    releaser = deployer.RELEASER();
+  }
+
+  function test_deploymentConfiguration() public view {
+    // Test TokenJar deployment and configuration
+    assertEq(tokenJar.releaser(), address(releaser), "Incorrect releaser on TokenJar");
+    assertEq(IOwned(address(tokenJar)).owner(), owner, "Incorrect owner on TokenJar");
+
+    // Test Releaser deployment and configuration
+    assertEq(address(releaser.RESOURCE()), RESOURCE, "Incorrect resource token");
+    assertEq(releaser.threshold(), THRESHOLD, "Incorrect threshold");
+    assertEq(address(releaser.TOKEN_JAR()), address(tokenJar), "Incorrect TokenJar address");
+    assertEq(releaser.thresholdSetter(), owner, "Incorrect threshold setter");
+    assertEq(IOwned(address(releaser)).owner(), owner, "Incorrect owner on Releaser");
+  }
+
+  function test_sequencerFeesAccumulation() public {
+    // Simulate sequencer fees being sent to TokenJar
+    uint256 initialBalance = address(tokenJar).balance;
+
+    // Send ETH to TokenJar (simulating sequencer fees)
+    uint256 feeAmount = 1 ether;
+    vm.deal(address(this), feeAmount);
+    (bool success,) = address(tokenJar).call{value: feeAmount}("");
+    assertTrue(success, "Failed to send ETH to TokenJar");
+
+    // Verify ETH accumulated in TokenJar
+    assertEq(
+      address(tokenJar).balance, initialBalance + feeAmount, "ETH not accumulated in TokenJar"
+    );
+  }
+
+  function test_releaseWithUNIBurn() public {
+    // Setup: Send sequencer fees to TokenJar
+    uint256 ethAmount = 5 ether;
+    vm.deal(address(this), ethAmount);
+    (bool success,) = address(tokenJar).call{value: ethAmount}("");
+    assertTrue(success, "Failed to send ETH to TokenJar");
+
+    // Deal UNI tokens to the caller for burning
+    address caller = address(0x1234);
+    deal(RESOURCE, caller, THRESHOLD);
+    assertEq(IERC20(RESOURCE).balanceOf(caller), THRESHOLD, "UNI not dealt to caller");
+
+    // Record balances before release
+    uint256 recipientBalanceBefore = address(0x5678).balance;
+    uint256 tokenJarBalanceBefore = address(tokenJar).balance;
+    uint256 uniSupplyBefore = IERC20(RESOURCE).totalSupply();
+
+    // Execute release (burning UNI to release ETH)
+    uint256 _nonce = releaser.nonce();
+    Currency[] memory currencies = new Currency[](1);
+    currencies[0] = Currency.wrap(address(0)); // ETH represented as address(0)
+
+    vm.startPrank(caller);
+    IERC20(RESOURCE).approve(address(releaser), THRESHOLD);
+    releaser.release(_nonce, currencies, address(0x5678));
+    vm.stopPrank();
+
+    // Verify ETH transferred from TokenJar to recipient
+    assertEq(address(tokenJar).balance, 0, "TokenJar should be empty");
+    assertEq(
+      address(0x5678).balance - recipientBalanceBefore,
+      tokenJarBalanceBefore,
+      "Incorrect ETH transferred to recipient"
+    );
+
+    // Verify UNI was burned
+    assertEq(
+      uniSupplyBefore - IERC20(RESOURCE).totalSupply(), THRESHOLD, "UNI not burned correctly"
+    );
```

Comment:
seems like the OP-bridged tokens decrement totalSupply for transfers to the bridge? super cool